pslogin/src/main/scala/WorldSessionActor.scala

```diff
@@ -3389,13 +3441,42 @@ class WorldSessionActor extends Actor with MDCContextAware {
     * @see `HackMessage`
     */
   //TODO add params here depending on which params in HackMessage are important
-  //TODO sound should be centered on IFFLock, not on player
   private def FinishHackingDoor(target : IFFLock, unk : Long)() : Unit = {
     target.Actor ! CommonMessages.Hack(player)
     localService ! LocalServiceMessage(continent.Id, LocalAction.TriggerSound(player.GUID, TriggeredSound.HackDoor, player.Position, 30, 0.49803925f))
     localService ! LocalServiceMessage(continent.Id, LocalAction.HackTemporarily(player.GUID, continent, target, unk))
   }
 
+  /**
+    * The process of hacking a terminal
+    * Pass the message onto the terminal and onto the local events system.
+    * @param target the `terminal` being hacked
+    * @param unk na;
+    *            used by `HackingMessage` as `unk5`
+    * @see `HackMessage`
+    */
+  //TODO add params here depending on which params in HackMessage are important
+  private def FinishHackingTerminal(target : Terminal, unk : Long)() : Unit = {
+    target.Actor ! CommonMessages.Hack(player)
+    localService ! LocalServiceMessage(continent.Id, LocalAction.TriggerSound(player.GUID, TriggeredSound.HackDoor, player.Position, 30, 0.49803925f))
+    localService ! LocalServiceMessage(continent.Id, LocalAction.HackTemporarily(player.GUID, continent, target, unk))
+  }
+
+  /**
+    * The process of hacking a locker
+    * Pass the message onto the locker and onto the local events system.
+    * @param target the `locker` being hacked
+    * @param unk na;
+    *            used by `HackingMessage` as `unk5`
+    * @see `HackMessage`
+    */
+  private def FinishHackingLocker(target : Locker, unk : Long)() : Unit = {
+    target.Actor ! CommonMessages.Hack(player)
+    localService ! LocalServiceMessage(continent.Id, LocalAction.TriggerSound(player.GUID, TriggeredSound.HackDoor, player.Position, 30, 0.49803925f))
```

Comment:
So hacking a locker (or a terminal) makes a door hacking sound? or is this just from being a copy of the door hacking function?

We should change the name of the `TriggeredSound` value if the former.  We could also probably combine these into a "simple hack" completion function by making the param `target : PlanetSideServerObject with Hackable`.

Reply:
Yeah currently it's the door sound. I should have put a note on that. The only reason I've left it like that is I honestly can't remember if that's the correct sound for everything being hacked or not.

If anyone can remember differently I can change it.